File: jung-io/src/test/java/edu/uci/ics/jung/io/graphml/parser/TestKeyElementParser.java
/*
 * Copyright (c) 2008, The JUNG Authors
 *
 * All rights reserved.
 *
 * This software is open-source under the BSD license; see either
 * "license.txt" or
 * https://github.com/jrtom/jung/blob/master/LICENSE for a description.
 */

package edu.uci.ics.jung.io.graphml.parser;

import org.junit.Assert;
import org.junit.Test;

import edu.uci.ics.jung.io.GraphIOException;
import edu.uci.ics.jung.io.graphml.Key;

public class TestKeyElementParser extends AbstractParserTest {

    @Test(expected= GraphIOException.class)
    public void testNoId() throws Exception {
        
        String xml = 
            "<key/>";
        
        readObject(xml);
    }

    @Test
    public void testId() throws Exception {
        
        String xml = 
            "<key id=\"d1\"/>";
        
        Key key = (Key) readObject(xml);
        Assert.assertNotNull(key);
        Assert.assertEquals("d1", key.getId());
        Assert.assertEquals(null, key.getDescription());
        Assert.assertEquals(null, key.getDefaultValue());
        Assert.assertEquals(null, key.getAttributeName());
        Assert.assertEquals(null, key.getAttributeType());
        Assert.assertEquals(Key.ForType.ALL, key.getForType());
    }

    @Test
    public void testDesc() throws Exception {
        
        String xml = 
            "<key id=\"d1\">" +
                "<desc>this is my key</desc>" +
            "</key>";
        
        Key key = (Key) readObject(xml);
        Assert.assertNotNull(key);
        Assert.assertEquals("d1", key.getId());
        Assert.assertEquals("this is my key", key.getDescription());
        Assert.assertEquals(null, key.getDefaultValue());
        Assert.assertEquals(null, key.getAttributeName());
        Assert.assertEquals(null, key.getAttributeType());
        Assert.assertEquals(Key.ForType.ALL, key.getForType());
    }

    @Test
    public void testDefault() throws Exception {
        
        String xml = 
            "<key id=\"d1\">" +
                "<default>yellow</default>" +
            "</key>";
        
        Key key = (Key) readObject(xml);
        Assert.assertNotNull(key);
        Assert.assertEquals("d1", key.getId());
        Assert.assertEquals(null, key.getDescription());
        Assert.assertEquals("yellow", key.getDefaultValue());
        Assert.assertEquals(null, key.getAttributeName());
        Assert.assertEquals(null, key.getAttributeType());
        Assert.assertEquals(Key.ForType.ALL, key.getForType());
    }

    @Test
    public void testAttrNameType() throws Exception {
        
        String xml = 
            "<key id=\"d1\" attr.name=\"myattr\" attr.type=\"double\">" +
            "</key>";
        
        Key key = (Key) readObject(xml);
        Assert.assertNotNull(key);
        Assert.assertEquals("d1", key.getId());
        Assert.assertEquals(null, key.getDescription());
        Assert.assertEquals(null, key.getDefaultValue());
        Assert.assertEquals("myattr", key.getAttributeName());
        Assert.assertEquals("double", key.getAttributeType());
        Assert.assertEquals(Key.ForType.ALL, key.getForType());
    }

    @Test
    public void testForNode() throws Exception {
        
        String xml = 
            "<key id=\"d1\" for=\"node\">" +
            "</key>";
        
        Key key = (Key) readObject(xml);
        Assert.assertNotNull(key);
        Assert.assertEquals("d1", key.getId());
        Assert.assertEquals(null, key.getDescription());
        Assert.assertEquals(null, key.getDefaultValue());
        Assert.assertEquals(null, key.getAttributeName());
        Assert.assertEquals(null, key.getAttributeType());
        Assert.assertEquals(Key.ForType.NODE, key.getForType());
    }

    @Test
    public void testForEdge() throws Exception {
        
        String xml = 
            "<key id=\"d1\" for=\"edge\">" +
            "</key>";
        
        Key key = (Key) readObject(xml);
        Assert.assertNotNull(key);
        Assert.assertEquals("d1", key.getId());
        Assert.assertEquals(null, key.getDescription());
        Assert.assertEquals(null, key.getDefaultValue());
        Assert.assertEquals(null, key.getAttributeName());
        Assert.assertEquals(null, key.getAttributeType());
        Assert.assertEquals(Key.ForType.EDGE, key.getForType());
    }

    @Test
    public void testForGraph() throws Exception {
        
        String xml = 
            "<key id=\"d1\" for=\"graph\">" +
            "</key>";
        
        Key key = (Key) readObject(xml);
        Assert.assertNotNull(key);
        Assert.assertEquals("d1", key.getId());
        Assert.assertEquals(null, key.getDescription());
        Assert.assertEquals(null, key.getDefaultValue());
        Assert.assertEquals(null, key.getAttributeName());
        Assert.assertEquals(null, key.getAttributeType());
        Assert.assertEquals(Key.ForType.GRAPH, key.getForType());
    }

    @Test
    public void testForAll() throws Exception {
        
        String xml = 
            "<key id=\"d1\" for=\"all\">" +
            "</key>";
        
        Key key = (Key) readObject(xml);
        Assert.assertNotNull(key);
        Assert.assertEquals("d1", key.getId());
        Assert.assertEquals(null, key.getDescription());
        Assert.assertEquals(null, key.getDefaultValue());
        Assert.assertEquals(null, key.getAttributeName());
        Assert.assertEquals(null, key.getAttributeType());
        Assert.assertEquals(Key.ForType.ALL, key.getForType());
    }
}

File: jung-io/src/test/kotlin/edu/uci/ics/jung/io/graphml/parser/TestKeyElementParser.kt
/*
 * Copyright (c) 2008, The JUNG Authors
 *
 * All rights reserved.
 *
 * This software is open-source under the BSD license; see either
 * "license.txt" or
 * https://github.com/jrtom/jung/blob/master/LICENSE for a description.
 */
package edu.uci.ics.jung.io.graphml.parser

import edu.uci.ics.jung.io.GraphIOException
import edu.uci.ics.jung.io.graphml.Key
import org.junit.Assert
import org.junit.Test

class TestKeyElementParser : AbstractParserTest() {

  @Test(expected = GraphIOException::class)
  fun testNoId() {
    val xml = "<key/>"
    readObject(xml)
  }

  @Test
  fun testId() {
    val xml = "<key id=\"d1\"/>"

    val key = readObject(xml) as Key
    Assert.assertNotNull(key)
    Assert.assertEquals("d1", key.id)
    Assert.assertEquals(null, key.description)
    Assert.assertEquals(null, key.defaultValue)
    Assert.assertEquals(null, key.attributeName)
    Assert.assertEquals(null, key.attributeType)
    Assert.assertEquals(Key.ForType.ALL, key.forType)
  }

  @Test
  fun testDesc() {
    val xml = "<key id=\"d1\">" + "<desc>this is my key</desc>" + "</key>"

    val key = readObject(xml) as Key
    Assert.assertNotNull(key)
    Assert.assertEquals("d1", key.id)
    Assert.assertEquals("this is my key", key.description)
    Assert.assertEquals(null, key.defaultValue)
    Assert.assertEquals(null, key.attributeName)
    Assert.assertEquals(null, key.attributeType)
    Assert.assertEquals(Key.ForType.ALL, key.forType)
  }

  @Test
  fun testDefault() {
    val xml = "<key id=\"d1\">" + "<default>yellow</default>" + "</key>"

    val key = readObject(xml) as Key
    Assert.assertNotNull(key)
    Assert.assertEquals("d1", key.id)
    Assert.assertEquals(null, key.description)
    Assert.assertEquals("yellow", key.defaultValue)
    Assert.assertEquals(null, key.attributeName)
    Assert.assertEquals(null, key.attributeType)
    Assert.assertEquals(Key.ForType.ALL, key.forType)
  }

  @Test
  fun testAttrNameType() {
    val xml = "<key id=\"d1\" attr.name=\"myattr\" attr.type=\"double\">" + "</key>"

    val key = readObject(xml) as Key
    Assert.assertNotNull(key)
    Assert.assertEquals("d1", key.id)
    Assert.assertEquals(null, key.description)
    Assert.assertEquals(null, key.defaultValue)
    Assert.assertEquals("myattr", key.attributeName)
    Assert.assertEquals("double", key.attributeType)
    Assert.assertEquals(Key.ForType.ALL, key.forType)
  }

  @Test
  fun testForNode() {
    val xml = "<key id=\"d1\" for=\"node\">" + "</key>"

    val key = readObject(xml) as Key
    Assert.assertNotNull(key)
    Assert.assertEquals("d1", key.id)
    Assert.assertEquals(null, key.description)
    Assert.assertEquals(null, key.defaultValue)
    Assert.assertEquals(null, key.attributeName)
    Assert.assertEquals(null, key.attributeType)
    Assert.assertEquals(Key.ForType.NODE, key.forType)
  }

  @Test
  fun testForEdge() {
    val xml = "<key id=\"d1\" for=\"edge\">" + "</key>"

    val key = readObject(xml) as Key
    Assert.assertNotNull(key)
    Assert.assertEquals("d1", key.id)
    Assert.assertEquals(null, key.description)
    Assert.assertEquals(null, key.defaultValue)
    Assert.assertEquals(null, key.attributeName)
    Assert.assertEquals(null, key.attributeType)
    Assert.assertEquals(Key.ForType.EDGE, key.forType)
  }

  @Test
  fun testForGraph() {
    val xml = "<key id=\"d1\" for=\"graph\">" + "</key>"

    val key = readObject(xml) as Key
    Assert.assertNotNull(key)
    Assert.assertEquals("d1", key.id)
    Assert.assertEquals(null, key.description)
    Assert.assertEquals(null, key.defaultValue)
    Assert.assertEquals(null, key.attributeName)
    Assert.assertEquals(null, key.attributeType)
    Assert.assertEquals(Key.ForType.GRAPH, key.forType)
  }

  @Test
  fun testForAll() {
    val xml = "<key id=\"d1\" for=\"all\">" + "</key>"

    val key = readObject(xml) as Key
    Assert.assertNotNull(key)
    Assert.assertEquals("d1", key.id)
    Assert.assertEquals(null, key.description)
    Assert.assertEquals(null, key.defaultValue)
    Assert.assertEquals(null, key.attributeName)
    Assert.assertEquals(null, key.attributeType)
    Assert.assertEquals(Key.ForType.ALL, key.forType)
  }
}
